keras/engine/training.py

```diff
@@ -44,17 +46,22 @@ def _standardize_input_data(data, names, shapes=None,
             the batch axis of the arrays matches the expected
             value found in `shapes`.
         exception_prefix: String prefix used for exception formatting.
+        input_tensors: A list of input tensor based data sources.
 
     # Returns
         List of standardized input arrays (one array per model input).
 
     # Raises
         ValueError: in case of improperly formatted user-provided data.
     """
+    if input_tensors is None:
+        input_tensors = []
     if not names:
         return []
     if data is None:
         return [None for _ in range(len(names))]
+    num_input_tensors = sum(getattr(x, 'is_placeholder', False) is False for x in input_tensors)
```

Comment:
If it is not a placeholder it is assumed to be an input tensor. I wrote it the way you say above myself before the unit test caught my mistake. :-)